File: Leet 2574/solution.cpp
#include <algorithm>
#include <cstdlib>

class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        std::vector<int> result(nums.size());
        int total = std::accumulate(nums.begin(), nums.end(), 0);
        
        int left_sum = 0;
        for(int i = 0; i < nums.size(); i++) {
            total -= nums[i];

            if (left_sum >= total) 
                result[i] = left_sum - total;
            else
                result[i] = total - left_sum;

            left_sum += nums[i];
        }

        return result;
    }
};

File: Leet 2574/solution.cs
public class Solution
{
    public int[] LeftRightDifference(int[] nums)
    {
        int rSum = 0;
        for (int i = 1; i < nums.Length; i++)
            rSum += nums[i];

        int lSum = 0;
        int[] result = new int[nums.Length];
        for (int i = 0; i < nums.Length; i++)
        {
            result[i] = int.Abs(lSum - rSum);
            lSum += nums[i];

            if (i + 1 < nums.Length)
                rSum -= nums[i + 1];
        }

        return result;
    }
}

class Program
{
    private static void Main()
    {
        int[] nums = [10, 4, 8, 3];
        Solution solution = new();
        int[] result = solution.LeftRightDifference(nums);

        Console.Write("[");
        foreach (var val in result)
        {
            Console.Write($" {val} ");
        }
        Console.WriteLine("]");
    }
}